### PR TITLE
Fixing /isbn import, publishers and product_group

### LINKED ISSUE #3195: /isbn endpoint fails import (no `product_type` set)
### Relevant url?

* Actual: 
* Expected: 

### Related files

openlibrary/core/vendor.py within AmazonAPI `serialize`

```
@@ -195,11 +182,15 @@ class AmazonAPI:
             'edition_num': (edition_info and edition_info.edition and
                             edition_info.edition.display_value),
             'publish_date': publish_date,
+            'product_group': (item_info.get('classifications', {})
+                              .get('product_group', {})
+                              .get('product_value')),
             'physical_format': (
                 item_info and item_info.classifications and
                 getattr(item_info.classifications.binding, 'display_value', '').lower()
             ),
         }
```

### PR DESCRIPTION
**Pull request by mekarpeles**
Closes #3195 

hotfix

### Technical

API throttling is still wonky in this version

### Testing

try importing a book which doesn't yet exist on OL (either isbn 10 or 13)

### Evidence

http://dev.openlibrary.org/isbn/9781258051372 this title was added

### Stakeholders

@hornc, @cdrini

### CODE REVIEW

**Comment by mekarpeles** [comment]
@tabshaikh reports missing error handling on API timeouts:
image

File: openlibrary/core/vendors.py [hunk 1]
```diff
@@ -102,29 +102,25 @@ def get_products(self, asins, serialize=False, marketplace='www.amazon.com',
...
         """
         item_ids = asins if type(asins) is list else [asins]
+        _resources = (self.RESOURCES[resources] if resources
+                      else self.RESOURCES['import'])
+        request = GetItemsRequest(partner_tag=self.tag,
+                                  partner_type=PartnerType.ASSOCIATES,
+                                  marketplace=marketplace,
+                                  item_ids=item_ids,
+                                  resources=_resources,
+                                  **kwargs)
```
Comment (mekarpeles) [inline openlibrary/core/vendors.py]:
```suggestion
        try:
            request = GetItemsRequest(partner_tag=self.tag,
                                      partner_type=PartnerType.ASSOCIATES,
                                      marketplace=marketplace,
                                      item_ids=item_ids,
                                     resources=_resources,
                                     **kwargs)
        except ??:
            ??   
```

Comment (mekarpeles) [reply openlibrary/core/vendors.py]: https://user-images.githubusercontent.com/978325/77669173-a9c46800-6f41-11ea-82ea-4a2ba53c6bd6.png

Comment (cclauss) [reply openlibrary/core/vendors.py]: from paapi5_python_sdk.rest import ApiException
[...]
except ApiException as e:
    log(e)
    print(e)
    wait and try again?

File: openlibrary/core/vendors.py [hunk 2]
```diff
@@ -102,29 +103,28 @@ def get_products(self, asins, serialize=False, marketplace='www.amazon.com',
...
+        _resources = (self.RESOURCES[resources] if resources
+                      else self.RESOURCES['import'])
+        try:
+            request = GetItemsRequest(partner_tag=self.tag,
+                                      partner_type=PartnerType.ASSOCIATES,
+                                      marketplace=marketplace,
+                                      item_ids=item_ids,
+                                     resources=_resources,
+                                     **kwargs)
+        except ApiException as e:
```
Comment (cclauss) [inline openlibrary/core/vendors.py]:
```suggestion
                                      resources=_resources,
                                      **kwargs)
        except ApiException:
```

File: openlibrary/core/vendors.py [hunk 3]
```diff
@@ -102,29 +103,28 @@ def get_products(self, asins, serialize=False, marketplace='www.amazon.com',
...
+            return None
```
Comment (tfmorris) [inline openlibrary/core/vendors.py]: This needs to, at a minimum, log the error and perhaps increment an error counter in stats. Silently ignoring without logging means that we'll never know how frequently it's occurring.

File: openlibrary/core/vendors.py [hunk 4]
```diff
@@ -101,29 +105,31 @@ def get_products(self, asins, serialize=False, marketplace='www.amazon.com',
...
-
         # Wait before doing the request
         wait_time = 1 / self.throttling - (time.time() - self.last_query_time)
         if wait_time > 0:
             time.sleep(wait_time)
         self.last_query_time = time.time()
 
+        item_ids = asins if type(asins) is list else [asins]
+        _resources = (self.RESOURCES[resources] if resources
+                      else self.RESOURCES['import'])
```
Comment (cclauss) [inline openlibrary/core/vendors.py:116]:
```suggestion
        _resources = self.RESOURCES[resources or 'import']
```

**Pull request merged**